src/metatrain/experimental/gap/utils/extract_selected_atoms.py

```diff
@@ -0,0 +1,41 @@
+from typing import List, Optional
+
+from metatensor.torch import Labels
+from metatensor.torch.atomistic import System
+
+
+def extract_selected_atoms(
```

Comment:
This is very wrong, sorry. Doing it this way, you will end up removing atoms that belong to a separate domain for domain decomposition and thus compute the wrong representation.

This will also make it so there are maybe pairs in the NL between atoms that no longer exist in positions/types.